src/LayerEditor/ui/tools/id_map.py

```diff
@@ -0,0 +1,54 @@
+"""Copied from bgem and modified"""
+
+class IdObject:
```

Comment:
I wasn't comfortable with IdMap inheriting dict since it is still possible to add items without using appropriate method.  That could lead to missing or inconsistent ids.